# Project 9 -- Kennedy Zapalac

**TA Help:** N/A
    
**Collaboration:** N/A

In [1]:
%sql sqlite:////anvil/projects/tdm/data/lahman/lahman.db

## Question 1

a. Write a query to find out who won the 2022 World Baseball Series from table `seriespost`?

b. For this champion team, please find out the home runs (hr) rate and runs batted in (rbi) rate in year 2022 from table batting. Round the rates to 2 decimals. You may get rates by

    - hr_rate = sum of home runs / sum of hits
    - rbi_rate = sum of runs batted in / sum of hits

In [2]:
%%sql
SELECT * FROM seriespost WHERE yearID = 2022 AND round = 'WS';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(2022, 'WS', 'HOU', 'NL', 'PHI', 'NL', 4, 2, 0)]

In [17]:
%%sql
SELECT 
    ROUND(CAST(SUM(HR) AS REAL)/SUM(H), 2) AS hr_rate, # must cast at least one of the inputs as a real number so that the output is a decimal rather than 0
    ROUND(CAST(SUM(RBI) AS REAL)/SUM(H), 2) AS rbi_rate 
FROM 
    batting 
WHERE 
    yearID = 2022 AND teamID = 'HOU'; 

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(0.16, 0.53)]

The Houston team won the 2022 world series and had a home run rate of 0.16 and a runs batted in rate of 0.53.

## Question 2
a. For the champion team from question 1, please write a query that counts the number of RBIs for each athlete in the champion team during year 2022, using the `batting` table. Display your output in ascending order.

b. Run the query again, but this time, display the output in descending order.

c. Which athlete has the highest RBIs in this question? Please provide the player’s `playerID`, along with their first name and last name, from the `people` table

In [28]:
%%sql
SELECT
    b.playerID,
    p.nameFirst,
    p.nameLast,
    b.RBI
FROM
    batting AS b
JOIN
    people AS p
ON
    b.playerID = p.playerID
WHERE
    teamID = 'HOU' AND yearID = 2022
ORDER BY
    b.RBI ASC;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('abreubr01', 'Bryan', 'Abreu', 0),
 ('baezpe01', 'Pedro', 'Baez', 0),
 ('bielabr01', 'Brandon', 'Bielak', 0),
 ('blancro01', 'Ronel', 'Blanco', 0),
 ('brownhu01', 'Hunter', 'Brown', 0),
 ('garcilu05', 'Luis', 'Garcia', 0),
 ('javiecr01', 'Cristian', 'Javier', 0),
 ('jonesta01', 'Taylor', 'Jones', 0),
 ('martise01', 'Seth', 'Martinez', 0),
 ('matonph01', 'Phil', 'Maton', 0),
 ('mcculla02', 'Lance', 'McCullers', 0),
 ('montera01', 'Rafael', 'Montero', 0),
 ('mushipa01', 'Parker', 'Mushinski', 0),
 ('nerishe01', 'Hector', 'Neris', 0),
 ('odorija01', 'Jake', 'Odorizzi', 0),
 ('pareden01', 'Enoli', 'Paredes', 0),
 ('perezjo01', 'Joe', 'Perez', 0),
 ('pressry01', 'Ryan', 'Pressly', 0),
 ('smithwi04', 'Will', 'Smith', 0),
 ('stanery01', 'Ryne', 'Stanek', 0),
 ('taylobl01', 'Blake', 'Taylor', 0),
 ('urquijo01', 'Jose', 'Urquidy', 0),
 ('valdefr01', 'Framber', 'Valdez', 0),
 ('verlaju01', 'Justin', 'Verlander', 0),
 ('diazya02', 'Yainer', 'Diaz', 1),
 ('goodrni01', 'Niko', 'Goodrum', 1),
 ('castrja01', 'Jason', 'Castro', 3),
 ('leeko01', 'Korey', 'Lee', 4),
 ('henslda01', 'David', 'Hensley', 5),
 ('matijjj01', 'J. J.', 'Matijevic', 5),
 ('sirijo01', 'Jose', 'Siri', 10),
 ('vazquch01', 'Christian', 'Vazquez', 10),
 ('meyerja02', 'Jake', 'Meyers', 15),
 ('dubonma01', 'Mauricio', 'Dubon', 16),
 ('mancitr01', 'Trey', 'Mancini', 22),
 ('brantmi02', 'Michael', 'Brantley', 26),
 ('diazal02', 'Aledmys', 'Diaz', 38),
 ('mccorch01', 'Chas', 'McCormick', 44),
 ('maldoma01', 'Martin', 'Maldonado', 45),
 ('gourryu01', 'Yuli', 'Gurriel', 53),
 ('altuvjo01', 'Jose', 'Altuve', 57),
 ('penaje02', 'Jeremy', 'Pena', 63),
 ('bregmal01', 'Alex', 'Bregman', 93),
 ('alvaryo01', 'Yordan', 'Alvarez', 97),
 ('tuckeky01', 'Kyle', 'Tucker', 107)]

In [29]:
%%sql
SELECT
    b.playerID,
    p.nameFirst,
    p.nameLast,
    b.RBI
FROM
    batting AS b
JOIN
    people AS p
ON
    b.playerID = p.playerID
WHERE
    teamID = 'HOU' AND yearID = 2022
ORDER BY
    b.RBI DESC;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('tuckeky01', 'Kyle', 'Tucker', 107),
 ('alvaryo01', 'Yordan', 'Alvarez', 97),
 ('bregmal01', 'Alex', 'Bregman', 93),
 ('penaje02', 'Jeremy', 'Pena', 63),
 ('altuvjo01', 'Jose', 'Altuve', 57),
 ('gourryu01', 'Yuli', 'Gurriel', 53),
 ('maldoma01', 'Martin', 'Maldonado', 45),
 ('mccorch01', 'Chas', 'McCormick', 44),
 ('diazal02', 'Aledmys', 'Diaz', 38),
 ('brantmi02', 'Michael', 'Brantley', 26),
 ('mancitr01', 'Trey', 'Mancini', 22),
 ('dubonma01', 'Mauricio', 'Dubon', 16),
 ('meyerja02', 'Jake', 'Meyers', 15),
 ('sirijo01', 'Jose', 'Siri', 10),
 ('vazquch01', 'Christian', 'Vazquez', 10),
 ('henslda01', 'David', 'Hensley', 5),
 ('matijjj01', 'J. J.', 'Matijevic', 5),
 ('leeko01', 'Korey', 'Lee', 4),
 ('castrja01', 'Jason', 'Castro', 3),
 ('diazya02', 'Yainer', 'Diaz', 1),
 ('goodrni01', 'Niko', 'Goodrum', 1),
 ('abreubr01', 'Bryan', 'Abreu', 0),
 ('baezpe01', 'Pedro', 'Baez', 0),
 ('bielabr01', 'Brandon', 'Bielak', 0),
 ('blancro01', 'Ronel', 'Blanco', 0),
 ('brownhu01', 'Hunter', 'Brown', 0),
 ('garcilu05', 'Luis', 'Garcia', 0),
 ('javiecr01', 'Cristian', 'Javier', 0),
 ('jonesta01', 'Taylor', 'Jones', 0),
 ('martise01', 'Seth', 'Martinez', 0),
 ('matonph01', 'Phil', 'Maton', 0),
 ('mcculla02', 'Lance', 'McCullers', 0),
 ('montera01', 'Rafael', 'Montero', 0),
 ('mushipa01', 'Parker', 'Mushinski', 0),
 ('nerishe01', 'Hector', 'Neris', 0),
 ('odorija01', 'Jake', 'Odorizzi', 0),
 ('pareden01', 'Enoli', 'Paredes', 0),
 ('perezjo01', 'Joe', 'Perez', 0),
 ('pressry01', 'Ryan', 'Pressly', 0),
 ('smithwi04', 'Will', 'Smith', 0),
 ('stanery01', 'Ryne', 'Stanek', 0),
 ('taylobl01', 'Blake', 'Taylor', 0),
 ('urquijo01', 'Jose', 'Urquidy', 0),
 ('valdefr01', 'Framber', 'Valdez', 0),
 ('verlaju01', 'Justin', 'Verlander', 0)]

In [30]:
%%sql
SELECT
    b.playerID,
    p.nameFirst,
    p.nameLast,
    b.RBI
FROM
    batting AS b
JOIN
    people AS p
ON
    b.playerID = p.playerID
WHERE
    teamID = 'HOU' AND yearID = 2022
ORDER BY
    b.RBI DESC
LIMIT 1;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('tuckeky01', 'Kyle', 'Tucker', 107)]

Kyle Tucker had the greatest RBI of 107 of rthe Houston team in 2022. His player ID is tuckeky01.

## Question 3
a. Write a query that finds how many times the athlete from question 2 attended All Star Games.

b. Write a query to find out who is the athlete that attended most All Star Games in the entire data set.

In [34]:
%%sql
SELECT COUNT(*) FROM allstarfull WHERE playerID = 'tuckeky01';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1,)]

In [41]:
%%sql
SELECT 
    playerID,
    COUNT(*) AS attendance 
FROM 
    allstarfull 
GROUP BY 
    playerID 
ORDER BY
    attendance DESC
LIMIT 1;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('mayswi01', 26)]

In [46]:
%%sql
SELECT
    playerID,
    nameFirst,
    nameLast
FROM
    people
WHERE 
    playerID = 'mayswi01';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('mayswi01', 'Willie', 'Mays')]

Kyle Tucker only appeared in 1 All Star Game. Willie Mays appeared in the most All Star Games in this dataset. He appeared 26 times.

## Question 4
a. Write a query that gets the average salary for each athlete in the database. Display your output in descending order. Limit the output to 100 rows (i.e., to the top 100 salaries).

In [51]:
%%sql
SELECT
    playerID,
    AVG(salary) as avg_salary
FROM
    salaries
GROUP BY
    playerID
ORDER BY
    avg_salary DESC
LIMIT 100;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('tanakma01', 22000000.0),
 ('rodrial01', 18109829.636363637),
 ('howarry01', 15525500.0),
 ('teixema01', 15305357.142857144),
 ('sabatcc01', 14576190.4),
 ('jeterde01', 13927268.05263158),
 ('fieldpr01', 13901318.181818182),
 ('cabremi01', 13457901.642857144),
 ('mauerjo01', 13232692.307692308),
 ('cespeyo01', 13165609.2),
 ('verlaju01', 12974090.909090908),
 ('pujolal01', 12752527.25),
 ('kurodhi01', 12580817.57142857),
 ('hamelco01', 12540000.0),
 ('hernafe02', 12529844.181818182),
 ('linceti01', 12444375.0),
 ('hollima01', 12040131.083333334),
 ('greinza01', 12025169.166666666),
 ('canoro01', 11806527.272727273),
 ('priceda01', 11693042.285714285),
 ('zambrca01', 11495000.0),
 ('beltrca01', 11432376.777777778),
 ('soriaal01', 11425714.285714285),
 ('kershcl01', 11301875.0),
 ('hamiljo03', 10969057.9),
 ('ramirma02', 10885672.05263158),
 ('santajo01', 10803523.333333334),
 ('crawfca02', 10699530.5),
 ('kempma01', 10660600.0),
 ('hallaro01', 10642261.857142856),
 ('zitoba01', 10541153.846153846),
 ('cainma01', 10431333.181818182),
 ('wrighda03', 10354447.666666666),
 ('ramirha01', 10333454.545454545),
 ('gonzaad01', 10324148.75),
 ('suzukic01', 10258217.6875),
 ('hunteto01', 10091470.588235294),
 ('pencehu01', 9848333.333333334),
 ('poseybu01', 9827963.0),
 ('delgaca01', 9753266.666666666),
 ('weaveje02', 9751500.0),
 ('fukudko01', 9700000.0),
 ('beltrad01', 9638947.368421054),
 ('heltoto01', 9637647.05882353),
 ('ethiean01', 9511200.0),
 ('utleych01', 9472692.153846154),
 ('vottojo01', 9453829.444444444),
 ('ellsbja01', 9358952.333333334),
 ('leeca01', 9341071.42857143),
 ('matsuhi01', 9250000.0),
 ('peavyja01', 9200876.5),
 ('oswalro01', 9195000.0),
 ('beckejo02', 9151201.846153846),
 ('martipe02', 9141224.0625),
 ('abreujo02', 9110889.0),
 ('wellsve01', 9016835.153846154),
 ('lackejo01', 8996011.857142856),
 ('riverma01', 8914834.210526315),
 ('ordonma01', 8898049.733333332),
 ('ortizda01', 8889583.333333334),
 ('berkmla01', 8857500.0),
 ('lestejo01', 8855550.0),
 ('reyesjo01', 8830076.923076924),
 ('leecl02', 8821275.0),
 ('morneju01', 8816363.636363637),
 ('buehrma01', 8784285.714285715),
 ('zimmery01', 8759200.0),
 ('darviyu01', 8750000.0),
 ('ramirar01', 8699117.647058824),
 ('dunnad01', 8668846.153846154),
 ('wainwad01', 8657909.090909092),
 ('scherma01', 8604357.125),
 ('bondsba01', 8556605.545454545),
 ('bagweje01', 8542267.933333334),
 ('tulowtr01', 8538100.0),
 ('burneaj01', 8486558.823529411),
 ('jonesch06', 8477606.65),
 ('gonzaca01', 8445754.857142856),
 ('mccanbr01', 8394560.545454545),
 ('werthja01', 8378064.285714285),
 ('guerrvl01', 8369430.333333333),
 ('johnsra05', 8359524.714285715),
 ('zimmejo02', 8344166.666666667),
 ('drewjd01', 8314745.230769231),
 ('uptonju01', 8195925.777777778),
 ('papeljo01', 8188412.0),
 ('wilsocj01', 8144500.2727272725),
 ('grandcu01', 8090454.545454546),
 ('pettian01', 8078377.411764706),
 ('martivi01', 8071276.923076923),
 ('mussimi01', 8029645.5),
 ('piazzmi01', 8011733.466666667),
 ('sheffga01', 8000407.142857143),
 ('markani01', 7902909.090909091),
 ('castiru01', 7847476.333333333),
 ('kawakke01', 7833666.0),
 ('riosal01', 7832272.7272727275),
 ('posadjo01', 7830566.666666667),
 ('phillbr01', 7806824.181818182),
 ('youngmi02', 7789116.416666667)]

The player with the ID 'tanakma01' has the greatest average salary in our dataset of \$220,00,000.

## Question 5

Now create your own query about a topic that you are interested in. Use at least one of the aggregation functions, such as min, max, count, or sum. Be sure to use group by and display the results in order with order by

My question: which team has the highest average home runs over the years?

In [56]:
%%sql
SELECT
    name,
    AVG(HR) as avg_hr
FROM 
    teams
GROUP BY
    name
ORDER BY
    avg_hr DESC
LIMIT 1;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('Colorado Rockies', 178.13333333333333)]

The Colorado Rockies have the highest average home runs over the years.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.